lib/fastly_nsq/listener.rb

```diff
@@ -1,59 +1,115 @@
+# frozen_string_literal: true
 require 'fastly_nsq/message'
+require 'fastly_nsq/manager'
+require 'fastly_nsq/safe_thread'
 
 module FastlyNsq
   class Listener
-    def self.listen_to(**args)
-      new(**args).go
+    include FastlyNsq::SafeThread
+
+    def self.listen_to(*args)
+      new(*args).go
     end
 
     def initialize(topic:, processor:, channel: nil, consumer: nil, **options)
-      @topic        = topic
-      @processor    = processor
       @consumer     = consumer || FastlyNsq::Consumer.new(topic: topic, channel: channel)
+      @done         = false
       @logger       = options.fetch :logger, FastlyNsq.logger
+      @manager      = options[:manager] || FastlyNsq::Manager.new
       @preprocessor = options[:preprocessor]
+      @processor    = processor
+      @thread       = nil
+      @topic        = topic
     end
 
-    def go(run_once: false)
-      exit_on 'INT'
-      exit_on 'TERM'
+    def identity
```

Comment:
it is used in test...
I know that sounds bad but I used it to test the `cloning bit`...